File: algorithm/All-In-One-Java/문제/week1/src/Problem1.java
/**
 * Problem 1: 자릿수 더하기
 * 난이도: 🟢 Lv.1
 * 주제: 기본 연산
 *
 * 문제: 양의 정수 n이 주어질 때, n의 각 자릿수를 모두 더한 값을 반환하세요.
 *
 * 입력: 930211
 * 출력: 16 (9+3+0+2+1+1)
 *
 * 핵심: 숫자를 문자열로 변환하거나, 나머지 연산으로 각 자릿수 추출
 */

public class Problem1 {

    static class Solution {
        public int solution(int n) {
            String str = String.valueOf(n);
            return str.chars().reduce(0, (acc, c) -> acc + Integer.parseInt(String.valueOf((char) c )));
        }
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println("=== Problem 1: 자릿수 더하기 ===\n");

        // 테스트 케이스
        int[][] tests = {
            {930211, 16},
            {12345, 15},
            {100, 1},
            {1, 1},
            {999, 27}
        };

        for (int[] test : tests) {
            int input = test[0];
            int expected = test[1];
            int result = sol.solution(input);
            String status = result == expected ? "✓" : "✗";
            System.out.println("Test " + input + ": " + result + " (기대값: " + expected + ") " + status);
        }
    }
}

File: algorithm/All-In-One-Java/문제/week5/src-js/Problem1.js
/**
 * Problem 1: 두 정수 사이의 합
 * 난이도: 🟢 Lv.1
 * 주제: 수학, 루프
 *
 * 문제: 두 정수 a, b가 주어질 때, a와 b 사이에 속한 모든 정수의 합을 반환하세요.
 *      (a == b 인 경우 a 반환, a > b 인 경우에도 동작해야 함)
 *
 * 입력: a = 3, b = 5
 * 출력: 12  (3 + 4 + 5)
 *
 * 핵심: 등차수열의 합 공식 또는 루프로 계산
 */

function solution(a, b) {
  // TODO: 풀이 작성
}

// === 테스트 ===
const tests = [
  { input: [3, 5], expected: 12 },
  { input: [3, 3], expected: 3 },
  { input: [5, 3], expected: 12 },
  { input: [-3, 3], expected: 0 },
  { input: [-5, -3], expected: -12 },
];

console.log("=== Problem 1: 두 정수 사이의 합 ===\n");

tests.forEach((t, i) => {
  const [a, b] = t.input;
  const result = solution(a, b);
  const ok = result === t.expected;
  console.log(
    `Test ${i + 1}: a=${a}, b=${b} → ${result} (기대값: ${t.expected}) ${ok ? "✓" : "✗"}`
  );
});
